app/src/main/kotlin/com/softteco/template/ui/feature/devicedashboard/devices/thermometer/ThermometerDashboardViewModel.kt

```diff
@@ -0,0 +1,210 @@
+package com.softteco.template.ui.feature.devicedashboard.devices.thermometer
+
+import androidx.annotation.StringRes
+import androidx.compose.runtime.Immutable
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.patrykandpatrick.vico.core.cartesian.data.CartesianValueFormatter
+import com.patrykandpatrick.vico.core.common.data.ExtraStore
+import com.softteco.template.R
+import com.softteco.template.data.base.error.Result
+import com.softteco.template.data.device.Thermometer
+import com.softteco.template.data.device.ThermometerRepository
+import com.softteco.template.ui.components.snackbar.SnackbarController
+import com.softteco.template.utils.AppDispatchers
+import com.softteco.template.utils.ChartUtils.aggregateByInterval
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import java.time.Duration
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.time.temporal.ChronoUnit
+import javax.inject.Inject
+
+val xToDateMapKeyLocalDateTime = ExtraStore.Key<Map<Float, LocalDateTime>>()
+const val ITEM_PLACER_COUNT = 4
+const val MIN_VALUES_INTERVAL = 1
+
+@HiltViewModel
+class ThermometerDashboardViewModel @Inject constructor(
+    private val thermometerRepository: ThermometerRepository,
+    private val snackbarController: SnackbarController,
+    private val appDispatchers: AppDispatchers,
+) : ViewModel() {
+    private val loading = MutableStateFlow(false)
+
+    private val fullTemperatureHistory = MutableStateFlow<Map<LocalDateTime, Float>>(
+        emptyMap()
+    )
+
+    private val fullHumidityHistory = MutableStateFlow<Map<LocalDateTime, Float>>(
+        emptyMap()
+    )
+
+    private val thermometer = MutableStateFlow(
+        Thermometer(
+            "111",
```

Comment:
I would avoid setting random mock data. 
We can set an empty string. 
Then we can check if a device is valid by checking whether the id is not empty. 
Also, Theoretically, it could be visible to a user.